### ISSUE TITLE
Index creators and contributors (1xx, 7xx)

### ISSUE THREAD

**Issue by banukutlu** [psu-libraries/psulib_traject#36]
In GitLab by @rkt6 on Sep 19, 2018, 10:42

Come up with methods for indexing creators and contributors for display and basic/advanced search (sort will probably need to be done separately). Attached file includes methods from several other Blacklight institutions.

[creators.rb](/uploads/f030a9337494ff06653f97838a1640bb/creators.rb)

This issue may be a challenge that has to be tackled in phases. We may want to work out basic method, test it, and refine from there.

Some samples from my testing:

```
## Display
to_field 'author_person_display', extract_marc('100aqbcdkj', trim_punctuation: true, alternate_script: false)
to_field 'author_corp_display', extract_marc('110abcdfgklnj', trim_punctuation: true, alternate_script: false)
to_field 'author_meeting_display', extract_marc('111abcdfgklnpqj', trim_punctuation: true, alternate_script: false)
to_field 'addl_author_display', extract_marc('700aqbcdjk:710abcdfgjkln:711abcdfgjklnpq', trim_punctuation: true, alternate_script: false)

## Vernacular field copies
to_field 'author_person_vern_display', extract_marc('100aqbcdkj', trim_punctuation: true, :alternate_script=>:only)
to_field 'author_corp_vern_display', extract_marc('110abcdfgklnj', trim_punctuation: true, :alternate_script=>:only)
to_field 'author_meeting_vern_display', extract_marc('111abcdfgklnpqj', trim_punctuation: true, :alternate_script=>:only)
to_field 'addl_author_vern_display', extract_marc('700aqbcdkj:710abcdfgklnj:711abcdfgklnpqj', trim_punctuation: true, :alternate_script=>:only)

## Are there things we can do with relators?
to_field 'relator', extract_marc('100e:110e:111e')
```

```
## Author Search

to_field 'author_t', extract_marc("100aqbcdk:110abcdfgkln:111abcdfgklnpq")
to_field 'author_addl_t', extract_marc("700aqbcdk:710abcdfgkln:711abcdfgklnpqj")
to_field "all_authors_facet", extract_marc("100abcdqj:110abcdj:111ancdj:700abcdqj:710abcdj:711ancdj", trim_punctuation: true)
 
```

**Comment by banukutlu** [comment]
In GitLab by @rkt6 on Sep 19, 2018, 13:56

changed the description

**Comment by banukutlu** [comment]
In GitLab by @rkt6 on Sep 20, 2018, 9:23

Fields edited:

- author_t
- author_addl_t

Field added search:

- all_authors_facet

Fields added display:

- author_person_display
- author_person_vern_display
- author_corp_display
- author_corp_vern_display
- author_meeting_display
- author_meeting_vern_display
- addl_author_display
- addl_author_vern_display


Fields removed display:

- author_display
- author_vern_display

(Blacklight we'll need to change these)

**Comment by banukutlu** [comment]
In GitLab by @bzk60 on Sep 20, 2018, 9:46 mentioned in issue https://github.com/psu-libraries/psulib_blacklight/issues/104

**Comment by banukutlu** [comment]
In GitLab by @cmd32 on Oct 2,2018, 2:55 closed via commit a0955b0f
